core/src/main/java/org/infinispan/interceptors/impl/CacheMgmtInterceptor.java

```diff
@@ -674,10 +684,81 @@ public long getAverageRemoveTimeNanos() {
       return counters.get(StripeB.removeTimesFieldUpdater) / removes;
    }
 
+   @ManagedAttribute(
+         description = "Approximate number of entries currently in the cache, including persisted and expired entries",
+         displayName = "Approximate number of entries"
+   )
+   public long getApproximateEntries() {
+      // Don't restrict the segments in case some writes used CACHE_MODE_LOCAL
+      IntSet allSegments = IntSets.immutableRangeSet(cacheConfiguration.clustering().hash().numSegments());
+      // Do restrict the segments when counting entries in shared stores
+      IntSet writeSegments;
+      if (distributionManager != null) {
+         writeSegments = distributionManager.getCacheTopology().getLocalWriteSegments();
+      } else {
+         writeSegments = allSegments;
+      }
+      long persistenceSize = CompletionStages.join(approximatePersistenceSize(allSegments, writeSegments));
+      return approximateTotalSize(persistenceSize, allSegments);
+   }
+
+   private CompletionStage<Long> approximatePersistenceSize(IntSet privateSegments, IntSet sharedSegments) {
+      return persistenceManager.running().approximateSize(AccessMode.PRIVATE, privateSegments)
+                               .thenCompose(privateSize -> {
+                                  if (privateSize >= 0)
+                                     return CompletableFuture.completedFuture(privateSize);
+
+                                  return persistenceManager.running().approximateSize(AccessMode.SHARED, sharedSegments);
```

Comment:
The idea is that private stores actually store the entries multiple times, so `approximateEntries()` should count all the copies.
Shared stores do not store each entry multiple times, so it would be wrong to count them once for every primary+backup owner.

The other alternative I considered was to only count shared store entries in `approximateEntriesUnique()`, but I thought that might be too surprising with the current stat names. It would have made more sense if I made `approximateEntries` do what `approximateEntriesUnique` does now, with `approximateEntriesLocal` counting the entries stored locally on each node and thus excluding shared stores. But then `approximateEntriesLocal` is jarring when used through `ClusterCacheStats` (which extends `Stats`). Maybe `approximateEntriesPrivate` would work?

Reply:
pls no more stats about entries :/

Reply:
>  Shared stores do not store each entry multiple times, so it would be wrong to count them once for every primary+backup owner.

ok, but the shared store is counting *all* segments for `approximateEntries`. 
if you put 1000 entries, all nodes will report `approximateEntries  = 1000`
It just does not seem correct to me.

Reply:
> pls no more stats about entries :/

I didn't mean to add more entries, I meant renaming
approximateEntries -> approximateEntriesLocal/approximateEntriesPrivate
approximateEntriesUnique -> approximateEntries

Reply:
> > Shared stores do not store each entry multiple times, so it would be wrong to count them once for every primary+backup owner.
> 
> ok, but the shared store is counting _all_ segments for `approximateEntries`. if you put 1000 entries, all nodes will report `approximateEntries = 1000` It just does not seem correct to me.

No, `sharedSegments` is `distributionManager.getCacheTopology().getLocalWriteSegments()` for dist caches.

Reply:
> No, sharedSegments is distributionManager.getCacheTopology().getLocalWriteSegments() for dist caches.

ah! my bad. When I read `approximatePersistenceSize(allSegments, writeSegments)`, I "assumed" `allSegments` were used by the shared store :)

Reply:
`approximateEntries` -> `approximateLocalEntries`
`approximateEntriesUnique` -> `approximateLocalEntriesUnique`
may I suggest ^ ? 
having the word `local` in the stat name makes everything clearer IMO.

Reply:
@pruivo I stayed away from "local" in the name because `Stats` and `ClusterCacheStats` must use the same method names to refer to local stats (in StatsImpl) and to cluster-wide stats (in ClusterCacheStatsImpl), and although `CacheMgmtInterceptor` doesn't implement the interface, the rule of thumb seems to be to use the same names.

I could keep the `approximateEntries` and `approximateEntriesUnique` names in `Stats` + rename the methods to `approximateLocalEntries` and `approximateLocalEntriesUnique` in `CacheMgmtInterceptor`, but I'm afraid the difference (which doesn't exist for other stats) might make the names more confusing instead of making them clearer.

OTOH, as you've noticed, Hot Rod's `ServerStatistics` already uses different names for the cluster-wide stats, and the REST API doesn't seem to provide cluster-wide stats at the moment, so maybe this is a good time to walk away from using the `Stats` interface and the same stat names for both local and cluster-wide stats.

Reply:
fair enough. I'll wait for @wburns green light before merging.